### PR TITLE
Tetron/weaponizer

### PR DESCRIPTION
**Pull request by tetron**

### CODE REVIEW

File: FF1Lib/Weapon.cs [hunk 1]
```diff
@@ -200,6 +200,420 @@ public void MagisizeWeapons(MT19337 rng, bool balanced)
...
+			    }
+
+			    // Need to special case: Defense, Katana, Xcal, Vorpal, Masa
+
+			    int specialPower = -1;
+			    if (spellIndex == 0xFF) {
+				int powerChance = rng.Between(1, 100);
+				if (tier == 1 && powerChance <= 20) {
+				    specialPower = rng.Between(0, powers.Length-1);
+				} else if (tier > 1 && powerChance <= 80) {
```
Comment (mhn65536) [inline FF1Lib/Weapon.cs:454]: Isn't that a little high?

File: FF1Lib/Weapon.cs [hunk 2]
```diff
@@ -200,6 +200,420 @@ public void MagisizeWeapons(MT19337 rng, bool balanced)
...
+				    break;
+				case 3:
+				    goldvalue *= goldvalue*1.5;
+				    break;
+				case 4:
+				    goldvalue *= goldvalue*2;
+				    break;
+			    }
+
+			    goldvalue = Math.Min(goldvalue, 65000);
```
Comment (mhn65536) [inline FF1Lib/Weapon.cs:571]: 65535

File: FF1Lib/Weapon.cs [hunk 3]
```diff
@@ -200,6 +200,420 @@ public void MagisizeWeapons(MT19337 rng, bool balanced)
...
+			    } else if (weaponItemId == Item.Katana) {
+				permissions = EquipPermission.Ninja;
+			    } else {
+				permissions = promotedPermissions[icon];
+				switch (tier) {
+				    case 1:
+					permissions |= unpromotedPermissions[icon];
+					break;
+				    case 2:
+					if (rng.Between(1, 100) < 50) {
```
Comment (mhn65536) [inline FF1Lib/Weapon.cs:587]: Will that be understandable to players?

**Review by mhn65536 — APPROVED** [review]
should be good enough i think

File: FF1Lib/Weapon.cs (hunk 1 above)
Comment (tetron) [reply FF1Lib/Weapon.cs:454]: This means 20% of common weapons will have extra powers, and 80% of rare weapons will.  In vanilla, 100% of rare weapons have extra powers, that's what makes them rare.  Are you suggesting it's too high for common items?

File: FF1Lib/Weapon.cs [hunk 4]
```diff
@@ -200,6 +200,420 @@ public void MagisizeWeapons(MT19337 rng, bool balanced)
...
+		    var defenseSwordSpells = new string[] { "RUSE", "INV2", "FOG2", "WALL" };
+
+		    var weaponNames = new List<string>();
+		    var weaponGfx = new List<int>();
+
+		    for (int tier = 4; tier > 0; tier--) {
+			var requireTypes = new List<int> { 0, 1, 2, 3, 5 };
+			for (int count = 0; count < tierList[tier-1].Count; ) {
+			    var weaponItemId = tierList[tier-1][count];
+			    int weaponIndex = weaponItemId - tierList[0][0];
```
[the remark this answers is not in the document]
Comment (tetron) [reply FF1Lib/Weapon.cs:355]: (int)Item.WoodenNunchucks is the first weapon, but I changed it to refer to it directly.

File: FF1Lib/Weapon.cs (hunk 1 above)
Comment (mhn65536) [reply FF1Lib/Weapon.cs:454]: 80% seemed a bit high, but with your explanation, it's good

**Pull request merged**